### PR TITLE
Fix variable show

### PR DESCRIPTION
**Pull request by b1llow**
# SQUASH ME

**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -6466,3 +6466,43 @@ RZ_API void rz_core_perform_auto_analysis(RZ_NONNULL RzCore *core, RzCoreAnalysi
 	rz_cons_break_pop();
 	RZ_FREE(debugger);
 }
+
+RZ_API char *rz_core_analysis_var_to_string(RZ_NONNULL RzCore *core, RZ_NONNULL RzAnalysisVar *var) {
```
Comment (XVilka) [inline librz/core/canalysis.c]: Please add ownership and Doxygen, also a macro

File: librz/include/rz_core.h [hunk 2]
```diff
@@ -1281,6 +1281,8 @@ RZ_API RZ_OWN RzList /*<RzRegItem *>*/ *rz_core_reg_filter_items_sync(RZ_NONNULL
 RZ_API void rz_core_cmd_show_analysis_help(RZ_NONNULL RzCore *core);
 RZ_API void rz_core_rtr_enable(RZ_NONNULL RzCore *core, const char *cmdremote);
 
+RZ_API char *rz_core_analysis_var_to_string(RZ_NONNULL RzCore *core, RZ_NONNULL RzAnalysisVar *var);
```
Comment (XVilka) [inline librz/include/rz_core.h]: Ownership here

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/analysis/var.c [hunk 3]
```diff
@@ -151,21 +151,51 @@ RZ_API RZ_BORROW RzAnalysisVar *rz_analysis_function_set_var(RzAnalysisFunction
...
- * Add or update a variable \p var to the given function \p fcn.
+ * Add or update a variable \p var to the given function \p fcn. Will not update existing DWARF variables
  *
  * \param fcn the function which the variable will belong to
  * \param var the variable to add or update
  * \param size \p var's type size
  * \return the created or updated variable, or NULL if the operation could not be completed
  */
-RZ_API RZ_BORROW RzAnalysisVar *rz_analysis_function_add_var(RzAnalysisFunction *fcn, RZ_OWN RzAnalysisVar *var,
+RZ_IPI RZ_BORROW RzAnalysisVar *rz_analysis_function_add_var_dwarf(RzAnalysisFunction *fcn, RZ_OWN RzAnalysisVar *var,
```
Comment (ret2libc) [inline librz/analysis/var.c]: I don't understand what it means that this function "will not update existing DWARF variables". Also, why the `_dwarf` in the function name?

Comment (b1llow) [reply librz/analysis/var.c]: https://github.com/rizinorg/rizin/blob/97737f6c12df2d0a51dd942224c0e2cc8aa8ddc0/librz/analysis/var.c#L153-L160

So this function is written specifically for this case, so it has a `_dwarf` suffix.

**Comment by XVilka** [comment]
@imbillow see
```
>>> MALLOC_PERTURB_=110 /private/var/folders/bc/yvvw2g2s4jlg9r8x7tf61xv800009d/T/woodpecker-local-2147515922/rizinorg/rizin/build/test/unit/test_type
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
stdout:
test_types_get_base_type_struct OK
test_types_get_base_type_union OK
test_types_get_base_type_enum OK
test_types_get_base_type_typedef OK
test_types_get_base_type_atomic OK
test_types_get_base_type_not_found OK
test_types_get_base_types OK
test_types_get_base_types_of_kind OK
test_type_as_string ERR
[XX] Fail at line 498: rz_type_as_string: expected int * const *[0][0][0][9], got int *const*[0][0][0][9].

test_type_as_pretty_string ERR
[XX] Fail at line 602: complex const pointer type is ugly: expected const char ** const * const c[4];, got const char **const*constc[4];.

test_enum_types OK
test_const_types OK
test_array_types OK
test_struct_func_types OK
test_struct_array_types ERR
[XX] Fail at line 898: type as string with multidimensional array of pointers: expected struct alb { const char *b; int * const *a[0][0][0][9]; }, got struct alb { const char *b; int *const*a[0][0][0][9]; }.

test_struct_identifier_without_specifier OK
test_union_identifier_without_specifier OK
test_edit_types ERR
[XX] Fail at line 993: rz_type_declaration_as_string: expected struct alb { const char *b; int * const *a[0][0][0][9]; }, got struct alb { const char *b; int *const*a[0][0][0][9]; }.

test_references OK
test_addr_bits OK
test_typedef_loop OK
test_struct_union_loop OK
test_struct_union_loop OK
test_path_by_offset_struct OK
test_path_by_offset_union OK
test_path_by_offset_array OK
test_path_by_offset_typedef OK

stderr:
ERROR: Atomic type "int"
```

**Comment by XVilka** [comment]
@imbillow please remove the extra spaces fix commit so that we could merge this, it can be fixed later since just a minor eye candy, while this PR fixes some major bugs.

**Review by XVilka — APPROVED** [review]

**Pull request merged**